app/views/core/SubscribeModal.coffee

```diff
@@ -145,6 +167,8 @@ module.exports = class SubscribeModal extends ModalView
       @render()
       jqxhr = if @basicCoupon?.code
         me.subscribe(token, {couponID: @basicCoupon.code})
+      else if product.get('name') is 'basic_subscription_yearly'
```

Comment:
I may be getting this wrong, but this logic seems to want to be first, to me. More specific, then less specific.

I can imagine `@basicCoupon` somehow having a value even if the product is actually `'basic_subscription_yearly'`. Clicking into a different sub first, a reload happening on the page, opening and closing a modal, the coupon logic changing, moving this to Vuex and having more initial data, or something unforeseen in the future. 

Any logical issues with making this the first `if` and moving the `if @basicCoupon?.code` to be the `else if`?

Reply:
100% agree.